### DISCUSSION TITLE
The button.grid() method is not affecting the position of the button

### DISCUSSION THREAD

**Discussion by TechPro424** [TomSchimansky/CustomTkinter#2043] [Q&A] [answered]
In my app, I'm creating a CTkButton and then using the `button.grid()` method to display it
Code: https://gist.github.com/TechPro424/903d81b448f4ca9eb06a4e7cb9efece8
But when I change the column value from 0 to 4, there is no change in the position of the button
Images: https://imgur.com/a/u8Ldk3q/

Using ctk 5.2.1 and python 3.11.5

Where am I going wrong?

**Comment by Akascape** [comment]
@TechPro424 
Use columnconfigure and rowconfigure

**Reply by TechPro424** [reply]
```py
app.grid_columnconfigure(10, weight=1)
button.grid(row = 0, column = 11, padx=20, pady=20)
```

This makes the button centered if the column is 10, stick to left if less than 10, and stick to right if greater than 10
any way to make it behave more like an x-y plane where if I give column = 11 it moves 1 units to the right or something?

**Reply by Akascape** [reply]
@TechPro424 Unable to understand what you exactly want, but use sticky parameter if you want stick any widget edge to a column/row. Learn the grid system properly.

**Reply by TechPro424** [reply]
I **don't** want it to stick
but it is sticking to the left or right if column is below or above 10 respectively

**Reply by Akascape** [reply]
@TechPro424 
That is happening because you only configured 10 th column `app.grid_columnconfigure(10, weight=1)`

You have to configure to all the required columns like this: `app.grid_columnconfigure((9,10,11), weight=1)`

**Reply by TechPro424** [reply, marked as answer]
Ok
so if I were using a lot of column, I would have to configure them all?
